### DISCUSSION TITLE
Selecting the PC Bluetooth peripheral

### DISCUSSION THREAD

**Discussion by alessandrotoniolo** [kevincar/bless#123] [Q&A] [answered]
Hi,

if in my PC I have more than one Bluetooth peripheral (e.g. the built in one and an extra USB dongle), is it possible to select the one I want to use with Bless?
If yes I'll be able to use the other one with Bleak and to develop a BLE test on PC simulating both central and peripheral devices, before using the real ones.

Thanks.

**Comment by vladak** [comment, marked as answer]
On Linux at least, when the name of the adapter is not specified, `hci0` will be used: https://github.com/kevincar/bless/blob/bd0bdf03825876d377bc016c07bc08b4b1c671d5/bless/backends/bluezdbus/dbus/utils.py#L79-L81

One can specify the controller by setting the `adapter` parameter of the `BlessServer` initializer, e.g.:
```python
server = BlessServer(name=my_service_name, loop=loop, adapter="hci1")
```
The list of adapters you can get via the `hcitool dev` command.

**Reply by alessandrotoniolo** [reply]
Thank you. Very useful.

**Comment by kevincar** [comment]
@alessandrotoniolo did @vladak 's solution work for you?

Your original idea of simulating both peripheral and server is interesting and may depend on how many independent BLE modules you have on your system. I would imagine this is feasible if you have more than one.

**Reply by alessandrotoniolo** [reply]
Yes, @vladak solution works. Unfortunately it works only for Linux, because Windows and MacOS don't allow more than one BT peripheral to be used at same time. But it's enough for now.
Thank you.
